File: duplicate-assert/java-junit/JavaParserTest_testAppendHiddenSingleLineCommentNodes.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class JavaParserTest extends AbstractModuleTestSupport {

    @Test
    public void testAppendHiddenSingleLineCommentNodes() throws Exception {
        final DetailAST root =
            JavaParser.parseFile(new File(getPath("InputJavaParserHiddenComments.java")),
                JavaParser.Options.WITH_COMMENTS);

        final Optional<DetailAST> singleLineComment = TestUtil.findTokenInAstByPredicate(root,
            ast -> ast.getType() == TokenTypes.SINGLE_LINE_COMMENT);
        assertTrue(singleLineComment.isPresent(), "Single line comment should be present");

        final DetailAST comment = singleLineComment.get();

        assertEquals(13, comment.getLineNo(), "Unexpected line number");
        assertEquals(0, comment.getColumnNo(), "Unexpected column number");
        assertEquals("//", comment.getText(), "Unexpected comment content");

        final DetailAST commentContent = comment.getFirstChild();

        assertEquals(TokenTypes.COMMENT_CONTENT, commentContent.getType(), "Unexpected token type");
        assertEquals(13, commentContent.getLineNo(), "Unexpected line number");
        assertEquals(2, commentContent.getColumnNo(), "Unexpected column number");
        assertTrue(commentContent.getText().startsWith(" inline comment"),
                "Unexpected comment content");
    }
}

File: duplicate-assert/python-unittest/JavaParserTest_testAppendHiddenSingleLineCommentNodes.py
import unittest

class JavaParserTest(unittest.TestCase):

    def test_append_hidden_single_line_comment_nodes(self):
        root = JavaParser.parse_file(File(get_path("InputJavaParserHiddenComments.java")),
                                     JavaParser.Options.WITH_COMMENTS)

        single_line_comment = TestUtil.find_token_in_ast_by_predicate(
            root, lambda ast: ast.get_type() == TokenTypes.SINGLE_LINE_COMMENT
        )
        self.assertTrue(single_line_comment.is_present(), "Single line comment should be present")

        comment = single_line_comment.get()

        self.assertEqual(13, comment.get_line_no(), "Unexpected line number")
        self.assertEqual(0, comment.get_column_no(), "Unexpected column number")
        self.assertEqual("//", comment.get_text(), "Unexpected comment content")

        comment_content = comment.get_first_child()

        self.assertEqual(TokenTypes.COMMENT_CONTENT, comment_content.get_type(), "Unexpected token type")
        self.assertEqual(13, comment_content.get_line_no(), "Unexpected line number")
        self.assertEqual(2, comment_content.get_column_no(), "Unexpected column number")
        self.assertTrue(comment_content.get_text().startswith(" inline comment"),
                        "Unexpected comment content")
